vm.h

```diff
@@ -46,4 +48,22 @@ void vm_hook_opcode_handler(vm_env *, int, vm_handler);
 
 void vm_run(vm_env *env);
 
+vm_seg_info *vm_new_seg_info(char *mem, size_t sz);
```

Comment:
I tend to believe it is the better way to go. But due to the lack of experience on these. Not sure how to divide this part, especially it involves source code change to the heart of the VM itself.